core/embed/rust/src/zcash_primitives/pallas/scalar.rs

```diff
@@ -0,0 +1,108 @@
+use core::ops::Deref;
+
+use pasta_curves::{
+    arithmetic::FieldExt,
+    group::ff::PrimeField,
+    pallas::{Point, Scalar},
+};
+
+use crate::micropython::{
+    ffi,
+    gc::Gc,
+    map::Map,
+    obj::Obj,
+    qstr::Qstr,
+    typ::Type,
+    util,
+    wrap::{Wrapable, Wrapped},
+};
+
+pub static SCALAR_TYPE: Type = obj_type! {
+    name: Qstr::MP_QSTR_Scalar,
+    locals: &obj_dict!(obj_map! {
+            Qstr::MP_QSTR_to_bytes => obj_fn_1!(scalar_to_bytes).as_obj(),
+            Qstr::MP_QSTR_is_not_zero => obj_fn_1!(scalar_is_not_zero).as_obj(),
```

Comment:
seeing the usages, it might be nicer to implement `MP_UNARY_OP_BOOL` instead of a named function?

Reply:
that's a good idea :bulb:

Reply:
23beeecc618cb39e0fc1baee36b5efff19700b33